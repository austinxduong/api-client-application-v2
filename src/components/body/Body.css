.Body {
    display: flex;
    flex-direction: column;
}

.Body section {
    display: flex;
    flex-direction: row;
}