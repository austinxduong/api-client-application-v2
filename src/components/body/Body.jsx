/* eslint-disable max-len */
import styles from './Body.css';
import React from 'react';
import PropTypes from 'prop-types';
import Buttons from 'prop-types';


const Body = ({ url, method, json, onSubmit, onChange }) => {

  return (
    <section className={styles.Body}>
      <form onSubmit={onSubmit}>
        <input type="text" name="url" placeholder="URL" value={url} onChange={onChange} />
        <section>
          <Buttons method={method} onChange={onChange} />
          <button>Process</button>
        </section>
        <textarea placeholder=" RAW JSON" name="json" value={json} onChange={onChange}></textarea>
      </form>
    </section>
  );
};

Body.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  json: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Body;

// on lines 15, we are passing our "method" and "onChange" states, that we defined in our container, into the Buttons child component
// now that we passed our child component "Buttons" into the parent component "Body", lets wire in our Body component into our high level container
