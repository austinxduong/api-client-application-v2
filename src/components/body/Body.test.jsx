import React from 'react';
import Body from '../body/Body';

import { render, cleanup } from '@testing-library/react';

describe('Body component', () => {
  afterEach(() => cleanup());
  it('Body component is functional', () => {
    const { asFragment } = render(<Body />);
    expect(asFragment()).toMatchSnapshot();
  });
});


