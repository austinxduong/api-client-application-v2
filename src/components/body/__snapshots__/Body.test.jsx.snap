// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body component Body component is functional 1`] = `
<DocumentFragment>
  <section
    class="Body"
  >
    <form>
      <input
        name="url"
        placeholder="URL"
        type="text"
        value=""
      />
      <section>
        <div>
          <input
            id="get"
            name="method"
            type="radio"
            value="GET"
          />
          <label
            for="get"
          >
             GET 
          </label>
          <input
            id="post"
            name="method"
            type="radio"
            value="POST"
          />
          <label
            for="post"
          >
             POST 
          </label>
          <input
            id="put"
            name="method"
            type="radio"
            value="PUT"
          />
          <label
            for="put"
          >
             PUT 
          </label>
          <input
            id="patch"
            name="method"
            type="radio"
            value="PATCH"
          />
          <label
            for="patch"
          >
             PATCH 
          </label>
          <input
            id="delete"
            name="method"
            type="radio"
            value="DELETE"
          />
          <label
            for="delete"
          >
             DELETE 
          </label>
        </div>
        <button>
           Process 
        </button>
      </section>
      <textarea
        name="json"
        placeholder=" RAW JSON"
      />
    </form>
  </section>
</DocumentFragment>
`;
